src/main.ts

```diff
@@ -18,18 +18,60 @@ import {groupDependenciesByManifest} from './utils'
 import {commentPr} from './comment-pr'
 import {getDeniedChanges} from './deny'
 
+async function delay(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function getComparison(
+  baseRef: string,
+  headRef: string,
+  retryOpts?: {
+    retryUntil: number
+    retryDelay: number
+  }
+): ReturnType<typeof dependencyGraph.compare> {
```

Comment:
(Q) why do we do this instead of specifying an explicit type?

Reply:
My guess is that it's because this is just a wrapper function, so passing through the type as well as the value could simplify things when refactoring.